streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -564,6 +579,13 @@ public void suspend() {
     void suspend(final boolean clean,
                  final boolean isZombie) {
         try {
+            if (clean) {
```

Comment:
Why not do this unconditionally? If it's not `clear` we won't commit anyway. It's seems cleaner to avoid to many branches and it's not on the hot code path so the overhead of updating `partitionTime` is not relevant.

Reply:
Alright, that's fine.